dlog/commitment/src/commitment.rs

```diff
@@ -125,51 +132,51 @@ fn shamir_sum<G: AffineCurve>(
 }
 
 impl<G: AffineCurve> SRS<G> {
-    // This function commits the polynomial against SRS instance with degree bound
-    //     plnm: polynomial to commit
-    //     max: maximal degree of the polynomial
-    //     RETURN: commitment group elements: unshifted, shifted
-    pub fn commit_with_degree_bound(
+    // This function commits a polynomial against URS instance
+    //     plnm: polynomial to commit to with max size of sections
+    //     max: maximal degree of the polynomial, if none, no degree bound
+    //     RETURN: tuple of: unbounded commitment vector, optional bounded commitment
+    pub fn commit
+    (
         &self,
         plnm: &DensePolynomial<Fr<G>>,
-        max: usize,
-    ) -> Result<(G, G), ProofError> {
-        let d = self.g.len();
-        if d < max || plnm.coeffs.len() > max {
-            return Err(ProofError::PolyCommit);
-        }
-
-        Ok((
-            self.commit_no_degree_bound(&plnm)?,
-            VariableBaseMSM::multi_scalar_mul(
-                &self.g[d - max..plnm.len() + d - max],
-                &plnm
-                    .coeffs
-                    .iter()
-                    .map(|s| s.into_repr())
-                    .collect::<Vec<_>>(),
-            )
-            .into_affine(),
-        ))
-    }
-
-    // This function commits the polynomial against SRS instance without degree bound
-    //     plnm: polynomial to commit
-    //     RETURN: commitment group element
-    pub fn commit_no_degree_bound(&self, plnm: &DensePolynomial<Fr<G>>) -> Result<G, ProofError> {
-        if self.g.len() < plnm.coeffs.len() {
-            return Err(ProofError::PolyCommit);
+        max: Option<usize>,
+    ) -> PolyComm<G>
+    {
+        PolyComm::<G>
+        {
+            // committing all the segments without shifting
```

Comment:
Can you separate some of the subexpressions out into separate bindings? I think it would make this a bit more readable. Like

```rust
let n = self.g.len();
let segments = (plnm.len() + (n-1)) / n;
let unshifted = (0..segments).map(|i| ...);
```

Reply:
This has been addressed.